### PR TITLE
Fix detected token count

### LINKED ISSUE #6116: Detected tokens number does not include tokens in protocols
## Problem Definition

The detected tokens number for an address does not necessarilly match the number of tokens we show below it. An example in the screenshot.

Says 2 tokens are detected, while you see 4. Which is confusing.
The reality is that indeed 2 tokens are detected as directly owned. The LQTY is in staking and the COMP in the comptroller yet to be claimed.

In the backend these are 2 cases of tokens we detect from protocols, when not directly owned via `balanceOf`.

## Task

This issue touches multiple points as it questions the location of tokens owned. We have discussed this in other issues too and probably a proper solution to token locations would help here. This would be the "right" way to solve this.

But perhaps we can also do something as a "temporary" fix to show the right number of tokens there? A frontend only solution counting the actual tokens shown after balance query?

### PR DESCRIPTION
**Pull request by tewshi**
Part of #6116 

## Checklist

- [ ] The PR modified the frontend, and updated the [user guide](https://github.com/rotki/rotki/blob/develop/docs/usage_guide.rst) to reflect the changes.

### CODE REVIEW

File: frontend/app/src/store/blockchain/balances/chains.ts [hunk 1]
```diff
@@ -39,11 +39,21 @@ export const useChainBalancesStore = defineStore('balances/chain', () => {
...
     set(balances, {
       ...get(balances),
-      [chain]: perAccount[chain] ?? {}
+      [chain]: chainValues
     });
 
+    // todo: this is temporary, to update the tokens count
+    // todo: remove when BE updates the endpoint to refresh detected tokens
+    if (chain === Blockchain.OPTIMISM) {
+      const { updateStateOnBalanceRefresh } = useBlockchainTokensStore();
```
Comment (kelsos) [inline frontend/app/src/store/blockchain/balances/chains.ts:52]: Can we move this outside the update function scope and into the store setup scope?

Initializing stores/composables outside of the setup can be problematic.

File: frontend/app/src/store/blockchain/balances/eth.ts [hunk 2]
```diff
@@ -115,11 +115,21 @@ export const useEthBalancesStore = defineStore('balances/eth', () => {
...
     set(balances, {
       ...get(balances),
-      [chain]: perAccount[chain] ?? {}
+      [chain]: chainValues
     });
 
+    // todo: this is temporary, to update the tokens count
+    // todo: remove when BE updates the endpoint to refresh detected tokens
+    if (chain === Blockchain.ETH) {
+      const { updateStateOnBalanceRefresh } = useBlockchainTokensStore();
```
Comment (kelsos) [inline frontend/app/src/store/blockchain/balances/eth.ts:128]: Same here

File: frontend/app/src/store/blockchain/balances/chains.ts (hunk 1 above)
Comment (tewshi) [reply frontend/app/src/store/blockchain/balances/chains.ts:52]: Okay, will do

File: frontend/app/src/store/blockchain/balances/eth.ts (hunk 2 above)
Comment (tewshi) [reply frontend/app/src/store/blockchain/balances/eth.ts]: done.

File: frontend/app/src/store/blockchain/balances/chains.ts (hunk 1 above)
Comment (tewshi) [reply frontend/app/src/store/blockchain/balances/chains.ts]: done.

**Review by LefterisJP — COMMENTED** [review]
Testing this I got an error just by opening the PR on the dashboard of a test account

```
Error at querying blockchain balances: updateStateOnBalanceRefresh is not a function
```

```
index.ts? [sm]:63 TypeError: updateStateOnBalanceRefresh is not a function
    at Object.update (chains.ts? [sm]:53:7)
    at pinia.mjs:1365:30
    at handleFetch (index.ts? [sm]:60:7)
    at async fetch (index.ts? [sm]:93:5)
    at async Promise.allSettled (:8080/index 7)
    at async fetchBlockchainBalances (index.ts? [sm]:109:7)
    at async Promise.allSettled (:8080/index 0)
    at async refreshAccounts (index.ts? [sm]:88:5)
    at async Promise.allSettled (:8080/index 1)
    at async fetch (index.ts? [sm]:115:5)
```

**Comment by tewshi** [comment]
> Testing this I got an error just by opening the PR on the dashboard of a test account
> 
> ```
> Error at querying blockchain balances: updateStateOnBalanceRefresh is not a function
> ```
> 
> ```
> index.ts? [sm]:63 TypeError: updateStateOnBalanceRefresh is not a function
>     at Object.update (chains.ts? [sm]:53:7)
>     at pinia.mjs:1365:30
>     at handleFetch (index.ts? [sm]:60:7)
>     at async fetch (index.ts? [sm]:93:5)
>     at async Promise.allSettled (:8080/index 7)
>     at async fetchBlockchainBalances (index.ts? [sm]:109:7)
>     at async Promise.allSettled (:8080/index 0)
>     at async refreshAccounts (index.ts? [sm]:88:5)
>     at async Promise.allSettled (:8080/index 1)
>     at async fetch (index.ts? [sm]:115:5)
> ```

This is fixed.

**Review by kelsos — APPROVED** [review]
thanks @tewshi lgtm

**Pull request merged**